server/network.go

```diff
@@ -72,27 +85,17 @@ func newNetwork(execPath string, rootDataDir string, whitelistedSubnets string,
 		nodeNames[i] = nodeName
 		cfg.NodeConfigs[i].Name = nodeName
 
-		// need to whitelist subnet ID to create custom VM chain
-		// ref. vms/platformvm/createChain
-		cfg.NodeConfigs[i].ConfigFile = fmt.Sprintf(`{
-	"network-peer-list-gossip-frequency":"250ms",
-	"network-max-reconnect-delay":"1s",
-	"public-ip":"127.0.0.1",
-	"health-check-frequency":"2s",
-	"api-admin-enabled":true,
-	"api-ipcs-enabled":true,
-	"index-enabled":true,
-	"log-display-level":"INFO",
-	"log-level":"%s",
-	"log-dir":"%s",
-	"db-dir":"%s",
-	"whitelisted-subnets":"%s"
-}`,
-			strings.ToUpper(logLevel),
-			logDir,
-			dbDir,
-			whitelistedSubnets,
-		)
+		// get the node configs right
+		nodeConfig := defaultNodeConfig
+		if nodeConfigParam != "" {
+			nodeConfig = nodeConfigParam
+		}
+		finalJSON, err := evalConfig(nodeConfig, logLevel, logDir, dbDir, whitelistedSubnets)
```

Comment:
this is string same as nodeConfig, probably call it finalNodeConfig instead of finalJSON.
also, evalConfig may not be the most clear name, probably replaceInConfig. Also, a good and more general
 function may be one that takes a map and replaces the map contents in the string.